pkg/controllers/trusted_ca_bundle_controller_test.go

```diff
@@ -0,0 +1,259 @@
+package controllers
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/util"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	systemCAValid   = "./fixtures/trust_bundle_valid.pem"
+	systemCAInvalid = "./fixtures/trust_bundle_invalid.pem"
+
+	additionalAmazonCAPemPath = "./fixtures/additional_ca_amazon.pem"
+	additionalMsCAPemPath     = "./fixtures/additional_ca_ms.pem"
+
+	// https://docs.openshift.com/container-platform/4.8/networking/configuring-a-custom-pki.html#nw-proxy-configure-object_configuring-a-custom-pki
+	additionalCAConfigMapName = "user-ca-bundle"
+	additionalCAConfigMapKey  = trustedCABundleConfigMapKey
+)
+
+func makeValidUserCAConfigMap(pemPath string) (*corev1.ConfigMap, error) {
+	testTrustBundle, err := ioutil.ReadFile(pemPath)
+	if err != nil {
+		return nil, err
+	}
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      additionalCAConfigMapName,
+			Namespace: OpenshiftConfigNamespace,
+		},
+		Data: map[string]string{
+			additionalCAConfigMapKey: string(testTrustBundle),
+		},
+	}, nil
+}
+
+func makeProxyResource() *v1.Proxy {
+	return &v1.Proxy{
+		ObjectMeta: metav1.ObjectMeta{Name: proxyResourceName},
+		Spec: v1.ProxySpec{
+			TrustedCA: v1.ConfigMapNameReference{Name: additionalCAConfigMapName},
+		},
+	}
+}
+
+var _ = Describe("Trusted CA bundle sync controller", func() {
+	var rec *record.FakeRecorder
+
+	var mgrCtxCancel context.CancelFunc
+	var mgrStopped chan struct{}
+	ctx := context.Background()
+
+	targetNamespaceName := testManagedNamespace
+
+	var reconciler *TrustedCABundleReconciler
+
+	var proxyResource *v1.Proxy
+	var additionalCAConfigMap *corev1.ConfigMap
+
+	mergedCAObjectKey := client.ObjectKey{Namespace: targetNamespaceName, Name: trustedCAConfigMapName}
+
+	BeforeEach(func() {
+		By("Setting up a new manager")
+		mgr, err := manager.New(cfg, manager.Options{MetricsBindAddress: "0"})
+		Expect(err).NotTo(HaveOccurred())
+
+		reconciler = &TrustedCABundleReconciler{
+			Client:          cl,
+			Scheme:          scheme.Scheme,
+			Recorder:        rec,
+			TargetNamespace: targetNamespaceName,
+			trustBundlePath: systemCAValid,
+		}
+		Expect(reconciler.SetupWithManager(mgr)).To(Succeed())
+
+		By("Creating needed ConfigMaps and Update Proxy")
+		proxyResource = makeProxyResource()
+		additionalCAConfigMap, err = makeValidUserCAConfigMap(additionalAmazonCAPemPath)
+		Expect(err).To(Succeed())
```

Comment:
This is more idiomatic in ginkgo/gomega (and matches line 76)
```suggestion
		Expect(err).NotTo(HaveOccurred())
```